Automattic/wp-calypso
# Jetpack pricing: make CTAs distinct

### Changes proposed in this Pull Request

The product cards in the [Jetpack pricing page](https://cloud.jetpack.com/pricing) have the same call-to-action (_Get_), which violate several accessibility guidelines. This PR append the product name.

### Implementation notes

Product names can potentially be formatted. Here's an example (though not in use anymore): 
```translate( 'Backup {{em}}Daily{{/em}}', { components: { em: <em/> } } )```.

That can't be used in HTML attributes (`aria-label` in our case) since it contains markup. To prevent devs from having to define product names twice, I've created a function that gets the string equivalent, in that example `Backup Daily`. That function is `reactNodeToString`.

I don't think there's a way to do that with the `react` or `react-dom` libs. There used to be a `renderToString` function, but it's deprecated and was only available server side, I believe.

### Testing instructions

- Spin up the pricing page, using a live link below, or by running this branch locally
- Using a screen reader, notice the _Get_ call-to-action in the product and bundle cards is read _Get [product name]_
- Alternatively, check that the call-to-action element has an `aria-label`, as seen below

<img width="1167" alt="Screen Shot 2022-12-12 at 4 24 26 PM" src="https://user-images.githubusercontent.com/1620183/207157594-89153ebb-62ef-48ff-bc98-8e389eb09f5d.png">

### Screenshots

Content of the _Links_ section, using VoiceOver's rotor on macOS.

_Before_

https://user-images.githubusercontent.com/1620183/207452147-41b1891c-8823-4c19-99be-0ad599964aa5.mov

_After_

https://user-images.githubusercontent.com/1620183/207452184-02a15c2a-12b2-4030-b2d6-1ec6bbdd3b83.mov
